ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/common/logging/NamedLogger.scala

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.common.logging
+
+import org.slf4j.{Logger, LoggerFactory}
+
+/**
+  * NamedLogger augments a regular class-based slf4j logger with one annotated with a "name" where the name provides
+  * human readable context identifying a class instance, e.g. which participant in a set of participants has logged
+  * a particular message.
+  *
+  * The name can be constructed in a nested, left-to-right append manner.
+  */
+trait NamedLogger {
+
+  /** Name for the logger. Can be empty. */
+  val name: String
+
+  /** Augment the name with another sub-name. Useful when transitioning from one caller to one specific named callee */
+  def append(subName: String): NamedLogger
+
+  def forParticipant(id: String): NamedLogger = append(s"participant/$id")
+
+  private[logging] def getLogger(fullName: String): Logger
+
+  /** get a loggers in factory methods
+    *
+    * Sometimes, the NamedLogging trait can not be used, e.g. in a factory method. In these
+    * cases, a logger can be created using this function.
+    */
+  def getLogger(klass: Class[_]): Logger = {
+    val fullName = Array(klass.getName, name)
+      .filterNot(_.isEmpty)
+      .mkString(":")
+    getLogger(fullName)
+  }
+
+}
+
+// Named Logger implementation
+private[logging] case class SimpleNamedLogger(name: String) extends NamedLogger {
```

Comment:
Does this really need to be a `case class`?
```suggestion
private[logging] final class SimpleNamedLogger(name: String) extends NamedLogger {
```

Reply:
Also worth considering calling this a `SimpleNamedLoggerFactory`.

Reply:
Changing to regular final class instead and renaming.